docs/source/user_guide/using_datasets.rst

```diff
@@ -2213,9 +2213,133 @@ of a sample with |Detections| below a specified threshold filtered out.
+:meth:`merge_samples() <fiftyone.core.dataset.Dataset.merge_samples>` method
+that you can use to merge the contents of another |Dataset| or |DatasetView|
+into an existing dataset.
+
+By default, samples with the same absolute `filepath` are merged, and top-level
+fields from the provided samples are merged in, overwriting any existing values
+for those fields, with the exception of list fields (e.g.,
+:ref:`tags <using-tags>`) and label list fields (e.g.,
+:ref:`Detections <object-detection>`), in which case the elements of the lists
+themselves are merged. In the case of label list fields, labels with the same
+`id` in both collections are updated rather than duplicated.
+
+The :meth:`merge_samples() <fiftyone.core.dataset.Dataset.merge_samples>`
+method can be configured in numerous ways, including:
+
+-   Which field to use as a merge key, or an arbitrary function defining the
+    merge key
+-   Whether existing samples should be modified or skipped
+-   Whether new samples should be added or omitted
+-   Whether new fields can be added to the dataset schema
+-   Whether list fields should be treated as ordinary fields and merged as a
+    whole rather than merging their elements
+-   Whether to merge only specific fields, or all but certain fields
+-   Mapping input fields to different field names of this dataset
+
+For example, the following snippet demonstrates merging a new field into an
+existing dataset:
 
-Batch actions
+.. code:: python
+    :linenos:
+
+    import fiftyone as fo
+    import fiftyone.zoo as foz
+
+    dataset1 = foz.load_zoo_dataset("quickstart")
+
+    # Create a dataset containing only ground truth objects
+    dataset2 = dataset1.select_fields("ground_truth").clone()
+
+    # Create a view containing only the predictions
+    predictions_view = dataset1.select_fields("predictions")
+
+    # Merge the predictions
+    dataset2.merge_samples(predictions_view)
+
+    print(dataset1.count("ground_truth.detections"))  # 1232
+    print(dataset2.count("ground_truth.detections"))  # 1232
+
+    print(dataset1.count("predictions.detections"))  # 5620
+    print(dataset2.count("predictions.detections"))  # 5620
+
+Note that the argument to
+:meth:`merge_samples() <fiftyone.core.dataset.Dataset.merge_samples>` can be a
+|DatasetView|, which means that you can perform possibly-complex
+:ref:`transformations <using-views>` to the source dataset to select the
+desired content to merge.
+
+Consider the following variation of the above snippet, which demonstrates a
+workflow where |Detections| from another dataset are merged into a dataset with
+existing |Detections| in the same field:
+
+.. code:: python
+    :linenos:
+
+    from fiftyone import ViewField as F
+
+    # Create a new dataset that only contains predictions with confidence >= 0.9
+    dataset3 = (
+        dataset1
+        .select_fields("predictions")
+        .filter_labels("predictions", F("confidence") > 0.9)
+    ).clone()
```

Comment:
I'll note that I get an error here when the `ground_truth` field has a similarity run.

```
ValueError                                Traceback (most recent call last)
<ipython-input-2-8295614b3f19> in <module>
      1 from fiftyone import ViewField as F
      2 # Create a new dataset that only contains predictions with confidence >= 0.9
----> 3 dataset3 = (
      4     dataset1
      5     .select_fields("predictions")

~/code/fiftyone/fiftyone/core/view.py in clone(self, name)
    634             the new :class:`Dataset`
    635         """
--> 636         return self._dataset._clone(name=name, view=self)
    637 
    638     def reload(self):

~/code/fiftyone/fiftyone/core/dataset.py in _clone(self, name, view)
   2077             sample_collection = self
   2078 
-> 2079         return _clone_dataset_or_view(sample_collection, name)
   2080 
   2081     def clear(self):

~/code/fiftyone/fiftyone/core/dataset.py in _clone_dataset_or_view(dataset_or_view, name)
   4338     # Clone RunResults
   4339     if dataset.has_evaluations or dataset.has_brain_runs:
-> 4340         _clone_runs(clone_dataset, dataset._doc)
   4341 
   4342     return clone_dataset

~/code/fiftyone/fiftyone/core/dataset.py in _clone_runs(dst_dataset, src_doc)
   4576     # memory and then writing a new copy for the destination dataset
   4577     for brain_key, run_doc in dst_doc.brain_methods.items():
-> 4578         results = fob.BrainMethod.load_run_results(dst_dataset, brain_key)
   4579         run_doc.results = None
   4580         fob.BrainMethod.save_run_results(dst_dataset, brain_key, results)

~/code/fiftyone/fiftyone/core/runs.py in load_run_results(cls, samples, key)
    441         except Exception as e:
    442             if run_doc.version == foc.VERSION:
--> 443                 raise e
    444 
    445             raise ValueError(

~/code/fiftyone/fiftyone/core/runs.py in load_run_results(cls, samples, key)
    438 
    439         try:
--> 440             run_results = RunResults.from_str(results_str, view)
    441         except Exception as e:
    442             if run_doc.version == foc.VERSION:

~/venvs/fon/lib/python3.8/site-packages/eta/core/serial.py in from_str(cls, s, *args, **kwargs)
    492             an instance of the Serializable class
    493         """
--> 494         return cls.from_dict(_load_json(s), *args, **kwargs)
    495 
    496     @classmethod

~/code/fiftyone/fiftyone/core/runs.py in from_dict(cls, d, samples)
    615 
    616         run_results_cls = etau.get_class(d["cls"])
--> 617         return run_results_cls._from_dict(d, samples)
    618 
    619     @classmethod

~/venvs/fon/lib/python3.8/site-packages/fiftyone/brain/similarity.py in _from_dict(***failed resolving arguments***)

~/venvs/fon/lib/python3.8/site-packages/fiftyone/brain/similarity.py in __init__(***failed resolving arguments***)

~/venvs/fon/lib/python3.8/site-packages/fiftyone/brain/similarity.py in _get_ids_for_embeddings(***failed resolving arguments***)

~/code/fiftyone/fiftyone/core/collections.py in _get_selected_labels(self, ids, tags, fields)
    701 
    702     def _get_selected_labels(self, ids=None, tags=None, fields=None):
--> 703         view = self.select_labels(ids=ids, tags=tags, fields=fields)
    704 
    705         labels = []

~/code/fiftyone/fiftyone/core/collections.py in select_labels(self, labels, ids, tags, fields, omit_empty)
   3968             a :class:`fiftyone.core.view.DatasetView`
   3969         """
-> 3970         return self._add_view_stage(
   3971             fos.SelectLabels(
   3972                 labels=labels,

~/code/fiftyone/fiftyone/core/view.py in _add_view_stage(self, stage)
    779 
    780     def _add_view_stage(self, stage):
--> 781         stage.validate(self)
    782 
    783         if stage.has_view:

~/code/fiftyone/fiftyone/core/stages.py in validate(self, sample_collection)
   4439             self._pipeline = self._make_labels_pipeline(sample_collection)
   4440         else:
-> 4441             self._pipeline = self._make_pipeline(sample_collection)
   4442 
   4443 

~/code/fiftyone/fiftyone/core/stages.py in _make_pipeline(self, sample_collection)
   4430         if self._omit_empty:
   4431             pipeline.extend(
-> 4432                 _make_omit_empty_labels_pipeline(sample_collection, fields)
   4433             )
   4434 

~/code/fiftyone/fiftyone/core/stages.py in _make_omit_empty_labels_pipeline(sample_collection, fields)
   5465     for field in fields:
   5466         match_exprs.append(
-> 5467             _get_label_field_only_matches_expr(sample_collection, field)
   5468         )
   5469 

~/code/fiftyone/fiftyone/core/stages.py in _get_label_field_only_matches_expr(sample_collection, field, prefix, new_field)
   5436     sample_collection, field, prefix="", new_field=None
   5437 ):
-> 5438     label_type = sample_collection._get_label_field_type(field)
   5439     is_label_list_field = issubclass(label_type, fol._LABEL_LIST_FIELDS)
   5440 

~/code/fiftyone/fiftyone/core/collections.py in _get_label_field_type(self, field_name)
   6360         if field_name not in schema:
   6361             ftype = "frame field" if is_frame_field else "field"
-> 6362             raise ValueError(
   6363                 "%s has no %s '%s'"
   6364                 % (self.__class__.__name__, ftype, field_name)

ValueError: DatasetView has no field 'ground_truth'
```

Reply:
ah I see. https://github.com/voxel51/fiftyone/issues/1167